### ISSUE TITLE
[Bug]: Fails when expected scheduled tasks are not found

### ISSUE THREAD

**Issue by SamErde** [HotCakeX/Harden-Windows-Security#334]
### Tools category

Harden Windows Security Module

### Does Your System Meet The Requirements?

- [X] Yes, I acknowledge that I've read the requirements and my system meets them. 👍

### Is your Windows Installation Genuine?

- [X] Yes, I acknowledge that the installation media of the Windows OS I used the tool on was downloaded from the official Microsoft website and I didn't tamper or modify it. 💯

### Did You Read The Frequently Asked Questions?

- [X] Yes, I've referred to the [FAQs](https://github.com/HotCakeX/Harden-Windows-Security/wiki/Answers-to-the-Basic-Frequently-Asked-Questions) and my issue is not covered/explained in there.

### Please Explain The Bug

🌜 I'll acknowledge right off that my "bug" was found while testing it on Windows Server 2025. That said, I do believe it could happen on other supported platforms if somebody manually removed a scheduled task. 🌛

The `XblGameSaveTask` scheduled task does not exist on this server. When I applied the hardening settings, it ran for a few seconds and then dumped the following exception back in the shell. At this point, the forms application became unusable and had to be closed by closing PowerShell.

I haven't peeked at your code yet, but trust that it would be easy to check for the existence of a scheduled task before attempting to act on it.

As an aside, it would be bonus if the GUI were able to show a message indicating that there was an error. 🌚

### Error Details  

`Exception calling "Run" with "1" argument(s): "ERROR: The specified task name "\Microsoft\XblGameSave\XblGameSaveTask" does not exist in the system.`

<details>
<summary>⚡Full Error and Logs Enclosed⚡</summary>

```powershell

PowerShell 7.4.5
PS C:\Users\Administrator> (irm 'https://raw.githubusercontent.com/HotCakeX/Harden-Windows-Security/main/Harden-Windows-Security.ps1')+'P'|iex
MethodInvocationException: Exception calling "Run" with "1" argument(s): "ERROR: The specified task name
"\Microsoft\XblGameSave\XblGameSaveTask" does not exist in the system. "


ErrorRecord                 : Exception calling "Run" with "1" argument(s): "ERROR: The specified
                              task name "\Microsoft\XblGameSave\XblGameSaveTask" does not exist in
                              the system.
                              "
WasThrownFromThrowStatement : False
TargetSite                  : Void CheckActionPreference(System.Management.Automation.Language.Functi
                              onContext, System.Exception)
Message                     : Exception calling "Run" with "1" argument(s): "ERROR: The specified
                              task name "\Microsoft\XblGameSave\XblGameSaveTask" does not exist in
                              the system.
                              "
Data                        : {[System.Management.Automation.Interpreter.InterpretedFrameInfo,
                              System.Management.Automation.Interpreter.InterpretedFrameInfo[]]}
InnerException              : System.Exception: ERROR: The specified task name
                              "\Microsoft\XblGameSave\XblGameSaveTask" does not exist in the system.

                                 at HardenWindowsSecurity.MicrosoftSecurityBaselines.Invoke()
                                 at HardenWindowsSecurity.GUIProtectWinSecurity.<>c.<AddEventHandlers
                              >b__0_18()
                                 at
                              System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread
                              threadPoolThread, ExecutionContext executionContext, ContextCallback
                              callback, Object state)
                              --- End of stack trace from previous location ---
                                 at
                              System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread
                              threadPoolThread, ExecutionContext executionContext, ContextCallback
                              callback, Object state)
                                 at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task&
                              currentTaskSlot, Thread threadPoolThread)
                              --- End of stack trace from previous location ---
                                 at HardenWindowsSecurity.GUIProtectWinSecurity.<>c.<<AddEventHandler
                              s>b__0_10>d.MoveNext()
                              --- End of stack trace from previous location ---
                                 at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_0(Object
                              state)
                                 at
                              System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate
                              callback, Object args, Int32 numArgs)
                                 at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object
                              source, Delegate callback, Object args, Int32 numArgs, Delegate
                              catchHandler)
                                 at System.Windows.Threading.DispatcherOperation.InvokeImpl()
                                 at
                              MS.Internal.CulturePreservingExecutionContext.CallbackWrapper(Object
                              obj)
                                 at System.Threading.ExecutionContext.RunInternal(ExecutionContext
                              executionContext, ContextCallback callback, Object state)
                              --- End of stack trace from previous location ---
                                 at System.Threading.ExecutionContext.RunInternal(ExecutionContext
                              executionContext, ContextCallback callback, Object state)
                                 at MS.Internal.CulturePreservingExecutionContext.Run(CulturePreservi
                              ngExecutionContext executionContext, ContextCallback callback, Object
                              state)
                                 at System.Windows.Threading.DispatcherOperation.Invoke()
                                 at System.Windows.Threading.Dispatcher.ProcessQueue()
                                 at System.Windows.Threading.Dispatcher.WndProcHook(IntPtr hwnd,
                              Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
                                 at
                              System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate
                              callback, Object args, Int32 numArgs)
                                 at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object
                              source, Delegate callback, Object args, Int32 numArgs, Delegate
                              catchHandler)
                                 at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPr
                              iority priority, TimeSpan timeout, Delegate method, Object args, Int32
                              numArgs)
                                 at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg,
                              IntPtr wParam, IntPtr lParam)
                                 at MS.Win32.UnsafeNativeMethods.DispatchMessage(MSG& msg)
                                 at
                              System.Windows.Threading.Dispatcher.PushFrameImpl(DispatcherFrame
                              frame)
                                 at System.Windows.Application.RunDispatcher(Object ignore)
                                 at System.Windows.Application.RunInternal(Window window)
                                 at CallSite.Target(Closure, CallSite, Application, Window)
HelpLink                    :
Source                      : System.Management.Automation
HResult                     : -2146233087
StackTrace                  :    at System.Management.Automation.ExceptionHandlingOps.CheckActionPref
                              erence(FunctionContext funcContext, Exception exception)
                                 at System.Management.Automation.Interpreter.ActionCallInstruction`2.
                              Run(InterpretedFrame frame)
                                 at System.Management.Automation.Interpreter.EnterTryCatchFinallyInst
                              ruction.Run(InterpretedFrame frame)
                                 at System.Management.Automation.Interpreter.EnterTryCatchFinallyInst
                              ruction.Run(InterpretedFrame frame)

MyCommand             :
BoundParameters       : {}
UnboundArguments      : {}
ScriptLineNumber      : 364
OffsetInLine          : 17
HistoryId             : 1
ScriptName            : C:\Program Files\WindowsPowerShell\Modules\Harden-Windows-Security-Module\0.5
                        .9\Core\Protect-WindowsSecurity.psm1
Line                  :                 [System.Void] [HardenWindowsSecurity.GUIMain]::app.Run([Harde
                        nWindowsSecurity.GUIMain]::mainGUIWindow)

Statement             : [System.Void] [HardenWindowsSecurity.GUIMain]::app.Run([HardenWindowsSecurity
                        .GUIMain]::mainGUIWindow)
PositionMessage       : At C:\Program Files\WindowsPowerShell\Modules\Harden-Windows-Security-Module\
                        0.5.9\Core\Protect-WindowsSecurity.psm1:364 char:17
                        + …             [System.Void] [HardenWindowsSecurity.GUIMain]::app.Run([H …
                        +               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
PSScriptRoot          : C:\Program
                        Files\WindowsPowerShell\Modules\Harden-Windows-Security-Module\0.5.9\Core
PSCommandPath         : C:\Program Files\WindowsPowerShell\Modules\Harden-Windows-Security-Module\0.5
                        .9\Core\Protect-WindowsSecurity.psm1
InvocationName        :
PipelineLength        : 0
PipelinePosition      : 0
ExpectingInput        : False
CommandOrigin         : Internal
DisplayScriptPosition :

PS C:\Users\Administrator>

```

</details>

**Comment by HotCakeX** [comment]
Just an update, Microsoft Security baselines trigger that error too, they do it in non-terminating way but i consider any error terminating to be on the safe side. I'll have to suppress it while applying the baseline.

![image](https://github.com/user-attachments/assets/e73494ff-f5ed-49e1-ad61-ff17eebc8c83)

**Comment by SamErde** [comment]
Cool. I like how that's a non-terminating error for non-essential objects, but you still clearly know when something isn't expected.

**Comment by HotCakeX** [comment]
Thanks again for reporting, i fixed it in version 0.6.0
https://github.com/HotCakeX/Harden-Windows-Security/releases/tag/Hardening-Module-v.0.6.0

Also as you suggested, I added a prompt when an error occurs that provides additional info and a button that links to this repo's issues section for easy reporting.

**Comment by SamErde** [comment]
Love it! Nice work.
